File: src/main/java/com/sm/carwashmonitor/controller/StatisticsController.java
package com.sm.carwashmonitor.controller;

import com.sm.carwashmonitor.dto.StationStatisticsDTO;
import com.sm.carwashmonitor.dto.StatisticsHighlightsDTO;
import com.sm.carwashmonitor.dto.StatisticsSummaryDTO;
import com.sm.carwashmonitor.service.StatisticsService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/statistics")
public class StatisticsController {

    private final StatisticsService statisticsService;

    @GetMapping("/highlights")
    public StatisticsHighlightsDTO getStatisticsHighlights(
            @RequestParam("dateTimeFrom") String dateTimeFrom,
            @RequestParam("dateTimeTo") String dateTimeTo,
            @RequestParam("timezone") String timezone
    ) {
        return statisticsService.getStatisticsHighlights(dateTimeFrom, dateTimeTo, timezone);
    }

    @GetMapping("/summary")
    public StatisticsSummaryDTO getStatisticsSummary(
            @RequestParam("dateTimeFrom") String dateTimeFrom,
            @RequestParam("dateTimeTo") String dateTimeTo,
            @RequestParam("timezone") String timezone
    ) {
        return statisticsService.getStatisticsSummary(dateTimeFrom, dateTimeTo, timezone);
    }
}

File: Controller/StatisticsController.cs
using CarwashMonitor.Dtos;
using CarwashMonitor.Service.Statistics;
using Microsoft.AspNetCore.Mvc;

namespace CarwashMonitor.Controller;

public class StatisticsController : ControllerBase
{
    private readonly IStatisticsService StatisticsService;

    public StatisticsController(IStatisticsService statisticsService)
    {
        StatisticsService = statisticsService;
    }

    [HttpGet]
    [Route("/statistics/highlights")]
    public async Task<ActionResult<StatisticsHighlightsDto>> GetStatisticsHighlightsAsync()
    {
        var result = await StatisticsService.GetStatisticsHighlightsAsync();
        return Ok(result);
    }

    [HttpGet]
    [Route("/statistics/summary")]
    public async Task<ActionResult<StatisticsSummaryDto>> GetStatisticsSummaryAsync()
    {
        var result = await StatisticsService.GetStatisticsSummaryAsync();
        return Ok(result);
    }
}